Add Ant Design

websemantics opened the pull request:
<!-- Please fill in the **bold** fields and tick all applicable boxes by placing an "x" inside "[ ]". -->

**https://github.com/websemantics/awesome-ant-design**

**Ant Design is a fairly new and popular UI framework (5k+ stars on Github) with tools, implementation ins in react and vue.js, for mobile and web apps. This works as an alternative to [Bootstrap](http://getbootstrap.com/), [Semantic-UI](http://semantic-ui.com/) or Github [Primer](https://github.com/primer/primer)**
# ALL THE BELOW CHECKBOXES SHOULD BE DONE AND CHECKED AND THE ABOVE PLACEHOLDERS SHOULD BE FILLED IN BEFORE SUBMITTING A PULL REQUEST!
- [x ] I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- [ x] This pull request has a descriptive title. _(For example: `Add Node.js`)_
- The list I added
  - [x ] **has been around for at least 20 days,**
  - [ x] is a non-generated Markdown file in a GitHub repo,
  - [ x] is not a duplicate,
  - [ x] only has awesome items with descriptions _(Awesome lists are curations of the best, not of everything)_,
  - [ x] contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
  - [ x] has a Table of Contents section named `Contents` as the first section,
  - [ x] has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) _(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)_,
  - [ x] has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
  - [x ] has been checked for inconsistent formatting, spelling, and grammar _(This includes starting each link description with an uppercase character and ending it with a period `.`, and consistent naming)_,
  - [ x] has been added at the bottom of the appropriate category.

aleksandar-todorovic commented:
Could you add the text of the license in the `LICENSE` file? 

Also, does this framework have an English version of their website?

websemantics commented:
- Added `LICENSE` file
- Yes, there is much effort to translate into English, http://beta.ant.design. And progress here, https://github.com/ant-design/ant-design/issues/1471
- I also added these links (English translation) to the repo README.md

Thanks @aleksandar-todorovic

davisonio commented:
> has a Table of Contents section named Contents as the first section,

Not named 'Contents'.

Some other feedback:

* Remove the 'other awesome lists' section. Two of them are outdated and unmaintained.
* Some of the descriptions are way too long in the 'related projects section'
* The list would look much better if the bullet points were not separated by lines

websemantics commented:
I just pushed the changes you highlighted above, 
I think the list does look nicer without the extra new lines, and long descriptions, 
Thanks for your feedback @davisonio ,

sindresorhus commented:
Feedback:
- Descriptions should be in the format `Link - Description.`. Note `-`, not `:` and all descriptions should start with an uppercase character.
- I would like to see more than just libraries. Include some blog posts, tutorials, links to community resources (Example: https://github.com/sindresorhus/awesome-electron#community), videos of talks, book, learning material, etc.
- Include project logo, like e.g. https://github.com/sindresorhus/awesome-electron
- There should be a succinct summary at the top about the project.
- This is a curation, which means you should only include awesome things. I see things in the list like "React Antd: React, Redux & Ant Design.", which says nothing about what it actually is, and I click the list and get an empty readme. Pretty useless. I realize it's a boilerplate, because I've seen so many before, but not everyone would.
- Be consistent about how you write different names of projects. I see both Vue and Vue.js, github and GitHub, Typescript and typescript (both are incorrect).
- Most descriptions could be improved.

websemantics commented:
Thanks for the feedback @sindresorhus  .. I'll get these rectified and update here ..

sindresorhus commented on readme.md:
```diff
@@ -184,6 +184,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
 - [Progressive Web Apps](https://github.com/TalAter/awesome-progressive-web-apps)
 - [choo](https://github.com/YerkoPalma/awesome-choo) - App framework.
 - [Redux](https://github.com/brillout/awesome-redux) - State container for JavaScript apps.
+- [Ant Design](https://github.com/websemantics/awesome-ant-design) - An enterprise-class UI design language.
```
Drop the `An`.

websemantics replied:
Well spotted, added to the todo list,

afc163 commented:
+1

sindresorhus commented:
ping

websemantics commented:
I think I covered most of the points you outlined, thanks again. I wasn't able to address some of the points however, like providing blog post, tutorials videos which I think will be more of as the community grows.

sindresorhus commented:
Can you fix the merge conflict? (Don't open a new PR, just update this one)

sindresorhus commented:
Also drop the `Inspired by the awesome list thing.`

-

> This is a curation, which means you should only include awesome things. I see things in the list like "React Antd: React, Redux & Ant Design.", which says nothing about what it actually is, and I click the list and get an empty readme. Pretty useless. I realize it's a boilerplate, because I've seen so many before, but not everyone would.

Still many user-unfriendly boilerplates without any readme. For example: https://github.com/okoala/react-redux-antd

websemantics commented:
@sindresorhus sorry for the delay and many thanks for your response, 

I remove the 'Inspired' paragraph and the link you referenced, 

I also move the boilerplates section to the bottom and I will continue to monitor and adjust it, 

I'm keeping it now as a reference for the community until it grows and better quality projects start to surface,

I updated the PR branch with the new content of `readme.md`, but it doesn't seem to resolve the conflict! I don't want to make it worst by pushing more changes, any idea what gone wrong?

sindresorhus commented:
> I'm keeping it now as a reference for the community until it grows and better quality projects start to surface,

None is better than low quality IMHO, but you should at least mark the low quality entries.

> I updated the PR branch with the new content of readme.md, but it doesn't seem to resolve the conflict! I don't want to make it worst by pushing more changes, any idea what gone wrong?

You could always start from scratch with a new branch, do the change, and then force push to this remote branch. (There's never a need to open a new pull request).

websemantics commented:
You're right! I actually checked all the links again and removed the low-quality repositories. Also renamed the section to Boilerplates. 

Tried to resolve this pull request conflict, but evidently unsuccessfully!

Ir1d commented:
@websemantics you can use 'git push -f'

websemantics commented:
Thanks @Ir1d, .. tried that, .. the branch [patch-1](https://github.com/websemantics/awesome/tree/patch-1) is up to date with  `readme.md` on `sindresorhus/awesome` master, but this PR still compares it with an older version it seems!

websemantics commented:
Ant Design trending today, https://github.com/trending
![ant-design](https://cloud.githubusercontent.com/assets/2190455/20715657/e585e962-b61d-11e6-8d44-dfce24dfb66f.png)

davisonio commented:
> Thanks @Ir1d, .. tried that, .. the branch patch-1 is up to date with readme.md on sindresorhus/awesome master, but this PR still compares it with an older version it seems!

Can you try it again? At the moment it says `This branch is 3 commits ahead, 22 commits behind sindresorhus:master.` Squashing the three commits together might make things easier https://github.com/sindresorhus/awesome/compare/master...websemantics:patch-1

websemantics commented:
Thanks @davisonio for the hint, ... This was a huge learning to get right or should I say green hh
